src/primval.rs

```diff
@@ -14,6 +14,8 @@ pub enum PrimVal {
 
     Ptr(Pointer),
     FnPtr(Pointer),
+    VtablePtr(Pointer, Pointer),
+    SlicePtr(Pointer, u64),
```

Comment:
I have tried out the `ByValPair` approach, and it definitely has advantages even now.